cpp/src_prims/sparse/utils.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+namespace MLCommon {
+namespace Sparse {
+
+template <typename value_idx>
+inline int block_dim(value_idx ncols) {
```

Comment:
maybe also look at the kernel's occupancy information to decide the best block-dim, along with `ncols`? IOW, query via cudart's [occupancy](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__OCCUPANCY.html#group__CUDART__OCCUPANCY) APIs to confirm if this particular block size will work.

Reply:
@teju85, are you referring to `cudaOccupancyMaxPotentialBlockSize`? Are  you thinking it should be used as-is or just as a preprocessing step for `ncols` before it is fed into through the conditionals below?

Reply:
@cjnolet never mind my above comment. That'd have probably complicated this logic for little perf gain (if any)!